### PR TITLE
Add ind_to_const to enable fewer equivalence classes.

### LINKED ISSUE #4963: parfors issue with recent patch
## Reporting a bug

This:
```python
from numba import njit
import numpy as np

N = 5
M = 17

@njit(parallel=True)
def test_func(nthreads):
    buf = np.zeros((M, N))
    return buf

test_func(23)
```
now reports 
```
numba/numba/lowering.py", line 1154, in getvar
    return self.varmap[name]
numba.errors.LoweringError: Failed in nopython mode pipeline (step: nopython mode backend)
'$12call_method.5_size0.1'

File "foo123321.py", line 9:
def test_func(nthreads):
    buf = np.zeros((M, N))
```
on master as of 71ebcc9.

bisect points to `[b7300f5f0e4c8b09cf58716101392c916b54d669] Experiment with union type` but it might be that in combination with other recent things.

### PR DESCRIPTION
**Pull request by DrTodd13**
Resolves #4963

### CODE REVIEW

**Comment by DrTodd13** [comment]
@stuartarchibald This patch seems to work for your test case and passes array analysis and parfor testing for me locally.  This is a moderately scary change.  Let me know what you think and how you want to progress.

**Comment by stuartarchibald** [comment]
Thanks for the patch @DrTodd13, much appreciated.
>  This is a moderately scary change. Let me know what you think and how you want to progress.

What are you most worried about? Getting the const var scopes right (local vs global)?

**Comment by DrTodd13** [comment]
> Thanks for the patch @DrTodd13, much appreciated.
> 
> > This is a moderately scary change. Let me know what you think and how you want to progress.
> 
> What are you most worried about? Getting the const var scopes right (local vs global)?

I'm not worried about scoping.  It is just a feeling about the degree of the change in combination with what triggered it which is new kinds of nodes being exposed in ways which previously weren't.  Anytime you add some new data structure to this alias analysis code I think that is a significant change.  I thought that might be a more isolated change than forcing both vars and consts into the ind_to_var dict.

File: numba/tests/test_parfors.py [hunk 1]
```diff
@@ -877,6 +880,13 @@ def test_impl(A):
         self.assertTrue(countParfors(test_impl, (types.Array(types.float64, 1, 'C'), )) == 2)
         self.assertTrue(countParfors(test_impl, (types.Array(types.float64, 2, 'C'), )) == 2)
 
+    @skip_unsupported
+    def test_issue4963(self):
+        def test_impl():
+            buf = np.zeros((_GLOBAL_INT_FOR_TESTING1, _GLOBAL_INT_FOR_TESTING2))
+            return buf
+        self.check(test_impl)
+
```
Comment (stuartarchibald) [inline numba/tests/test_parfors.py]: I think it'd be good to also add a freevar variant:
```python
    _FREEVAR_INT_FOR_TESTING1 = 17
    _FREEVAR_INT_FOR_TESTING2 = 5

    @skip_unsupported
    def test_issue4963_freevars(self):
        def test_impl():
            buf = np.zeros((_FREEVAR_INT_FOR_TESTING1, _FREEVAR_INT_FOR_TESTING2))
            return buf
        self.check(test_impl)
```

**Comment by stuartarchibald** [comment]
@DrTodd13 Added https://github.com/IntelLabs/numba/pull/65 which implements the above, if you are happy with it please merge and this PR can be considered complete! Thanks.

File: numba/tests/test_parfors.py (hunk 1 above)
Comment (DrTodd13) [reply numba/tests/test_parfors.py]: Merged.

Comment (stuartarchibald) [reply numba/tests/test_parfors.py]: Thanks!

**Pull request merged**